storm-server/src/main/java/org/apache/storm/daemon/supervisor/BasicContainer.java

```diff
@@ -622,6 +620,12 @@ protected String javaCmd(String cmd) {
         List<String> classPathParams = getClassPathParams(stormRoot, topoVersion);
         List<String> commonParams = getCommonParams();
 
+        String log4jConfigurationFile = getWorkerLoggingConfigFile();
```

Comment:
Nit: Can this go in `getCommonParams`?

Reply:
No, it can't -- The xml file might be in the topology jar uploaded to the nimbus. That file is not on the classpath if the LogWriter feature is used. 

So I guess it could, but getCommonParams would have to take a parameter and get invoked twice. Right now it's just invoked once and the output is used for both LogWriter and Worker jvm startup scripts.

Reply:
Thanks for explaining.